Mathlib/CategoryTheory/Sites/EffectiveEpimorphic.lean

```diff
@@ -0,0 +1,417 @@
+/-
+Copyright (c) 2023 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+
+import Mathlib.CategoryTheory.Sites.Sieves
+import Mathlib.CategoryTheory.Limits.Shapes.KernelPair
+
+/-!
+
+# Effective epimorphisms
+
+We define the notion of effective epimorphic (pre)sieves, morphisms and family of morphisms
+and provide some API for relating the three notions.
+
+-/
+
+namespace CategoryTheory
+
+open Limits
+
+variable {C : Type _} [Category C]
+
+/-- A sieve is effective epimorphic if the associated cocone is a colimit cocone. -/
```

Comment:
Can you discuss the relation with, for instance, https://stacks.math.columbia.edu/tag/00WP

Reply:
Added some comments about this in the module docstring.